projects/GKCore/GKCore/Controllers/PersonEditDlgController.cs

```diff
@@ -340,16 +340,13 @@ private void SetTarget(GDMIndividualRecord value)
                 if (fTarget != null) {
                     INamesTable namesTable = AppHost.NamesTable;
 
-                    var parts = GKUtils.GetNameParts(fBase.Context.Tree, fTarget);
+                    var parts = GKUtils.GetNameParts(fBase.Context.Tree, fTarget, false);
                     ICulture culture = parts.Culture;
-                    fView.Surname.Text = parts.Surname;
                     GDMSex sx = (GDMSex)fView.SexCombo.SelectedIndex;
 
                     switch (fTargetMode) {
                         case TargetMode.tmParent:
-                            if (sx == GDMSex.svFemale) {
-                                SetMarriedSurname(culture, parts.Surname);
-                            }
+                            fView.Surname.Text = parts.Surname;

```

Comment:
Here we're creating a child. All children have a surname of parents (in fact it is always a father)